en/21/06.md

```diff
@@ -0,0 +1,199 @@
+          fn zombies_count(&self) -> SingleValueMapper<usize>;
+
+          #[storage_mapper("zombies")]
+          fn zombies(&self, id: usize) -> SingleValueMapper<Zombie<Self::Api>>;
+        }
+    answer: |
+      #![no_std]
+
+      multiversx_sc::imports!();
+      multiversx_sc::derive_imports!();
+
+      #[derive(NestedEncode, NestedDecode, TopEncode, TopDecode, TypeAbi)]
+      pub struct Zombie<M: ManagedTypeApi> {
+          name: ManagedBuffer<M>,
+          dna: u64,
+      }
+
+      #[multiversx_sc::contract]
+      pub trait ZombiesContract {
+
+        #[init]
+        fn init(&self) {
+          self.dna_digits().set(16u8);
+          self.zombies_count().set(1usize);
+        }
+
+        fn create_zombie(&self, name: ManagedBuffer, dna: u64) {
+            self.zombies_count().update(|id| {
+              self.zombies(id).set(Zombie { name, dna });
+              *id +=1;
+            });
+        }
+
+        #[storage_mapper("dna_digits")]
+        fn dna_digits(&self) -> SingleValueMapper<u8>;
+        
+        #[storage_mapper("zombies_count")]
+        fn zombies_count(&self) -> SingleValueMapper<usize>;
+
+        #[storage_mapper("zombies")]
+        fn zombies(&self, id: &usize) -> SingleValueMapper<Zombie<Self::Api>>;
+      }
+---
+
+## Functions in Rust
+
+A function declaration in Rust looks like the following:
+
+```
+pub fn eat_hamburgers(&self, amount: u32) {
+
+}
+```
+
+This is a function named `eat_hamburgers` that takes 2 parameters: a reference to `self` and a `u32`. For now the body of the function is empty. Note that we're specifying the function visibility as `public`. Without the `pub` keyword the function would be private, visible only within the entity that implements it.
+
+What is a reference type you ask?
+
+Well, there are three ways in which you can pass an argument to a Rust function:
+
+ * By value, which means that the ownership over the parameter's value is passed to your function. This blocks the use of the initial parameter once the function call had ended since having it consummed by the function.
+ * By reference, which means that your function is borrowing the value of the parameter. Thus, this doesn't allow the function to change the value of the parameter since it is just `borrowed`. Once the function call finished, the ownership of the parameter returns to its initial owner.
+ * By mutable reference, which means that your function is borrowing the value of the parameter and gets the permission to change it. Once the function call finished, the ownership of the parameter returns to its initial owner keeping all changes done inside the function.
+
+```
+pub fn my function(value: u32, reference: &u32, mutable_reference: &mut u32) {
+  ...
+}
+```
+
+> Note: It's convention (but not required) to start function parameter variable names with an underscore (`_`) if the parameter is not used within the function. We'll use that convention throughout our tutorial.
```

Comment:
Sorry, but I don't follow. Why would you pass a parameter that's not used? Also, how is this related to the current topic?

Reply:
adapted from solidity but no longer required